pandas/tests/io/formats/test_format.py

```diff
@@ -305,6 +305,44 @@ def test_repr_non_interactive(self):
             assert not has_truncated_repr(df)
             assert not has_expanded_repr(df)
 
+    def test_repr_truncates_terminal_size(self):
+        # https://github.com/pandas-dev/pandas/issues/21180
+        # TODO: use mock fixutre.
+        # This is being backported, so doing it directly here.
+        try:
+            from unittest import mock
```

Comment:
should PR  in 0.24.0 (to move this to test_decorators)

Reply:
Forgot to lay out my plan.

I'd like to merge this with the `try / except`, backport to 0.23.2, and then make a PR removing this and using the `mock` fixture from https://github.com/pandas-dev/pandas/pull/20729

#20729 isn't being backported, so that seems easiset.